unitary/examples/quantum_rpg/qaracter.py

```diff
@@ -0,0 +1,140 @@
+from typing import Callable, Dict, List, Optional, Union
+import random
+from unitary import alpha
+from unitary.examples.quantum_rpg import enums
+
+
+class Qaracter(alpha.QuantumWorld):
+    """Base class for quantum RPG characters.
+
+  Quantum characters, or `Qaracter`s are made of
+  `QuantumObject`s.  The level of a quantum character
+  is equivalent to the number of `QuantumObject`s
+  (not including any ancillae added for measurement post-selection).
+
+  Qaracters have one hit point (HP) per level.  Qaracters are
+  considered DOWN if they have over half their qubits measured
+  as zero.  Qaracters are considered ESCAPED if they have all
+  their qubits measured and at least half are measured as one.
+  If neither of these is true, the Qaracter is considered to
+  be ACTIVE.
+
+  Additional HP can be added with the add_hp() function.
+  Additional quantum effects can be added with the add_quantum_effects()
+  function.
+
+  Hit point values can be sampled (a random result pulled from
+  the simulated distribution without measuring, which is possible
+  only in simulation) with sample(hp_num, False).
+
+  Hit point values can be measured by calling sample(hp_num, True).
+
+
+  Args:
+      name: name of this character as a string.
+  """
+
+    def __init__(self, name: str):
+        super().__init__()
+        self.name = name
+        self.health_status: Dict[alpha.QuantumObject, int] = {}
+        self.level = 0
+        self.add_hp()
+
+    def is_npc(self) -> bool:
+        """Returns Trus if a non-player or False if a player.
+
+    Inheritors of NPCs should override this function.
+    """
+        return False
+
+    def quantum_object_name(self, hp_num: int) -> str:
+        """Canonical name of a QuantumObject for this Qaracter.
+
+      Represents the n-th HP of a Qaracter.
+
+      Args:
+          hp_num: Number of the health point requested.  Health points
+              start at 1 and go up to the Qaracter's level (inclusive).
+      """
+        return f'{self.name}_{hp_num}'
+
+    def add_hp(self) -> alpha.QuantumObject:
+        """Adds an additional Health Point (HP to the Qaracter.
+
+      Each HP added to the Qaracter is an additional QuantumObject
+      with no effects (i.e. is a qubit with no operations).
+      Adding a HP also increases the Qaracter's level.
+      """
+        self.level += 1
+        obj = alpha.QuantumObject(self.quantum_object_name(self.level),
+                                  enums.HealthPoint.HURT)
```

Comment:
I don't know anything about the intended gameplay, but wouldn't HEALTHY be a more intuitive default for a newly added HP?

Reply:
Possibly, but I think it makes more sense from a quantum perspective to start at 0.  The 'qaracter's character sheet corresponds to quantum circuit that acts on a 'blank sheet' of all 0 HPs.